botorch/fit.py

```diff
@@ -121,12 +121,22 @@ def fit_gpytorch_model(
             if retry > 0:  # use normal initial conditions on first try
                 mll.model.load_state_dict(original_state_dict)
                 sample_all_priors(mll.model)
-            mll, _ = optimizer(mll, track_iterations=False, **kwargs)
+            try:
+                mll, _ = optimizer(mll, track_iterations=False, **kwargs)
+            except NotPSDError:
+                retry += 1
+                logging.log(
+                    logging.DEBUG,
+                    f"Fitting failed on try {retry} due to a NotPSDError.",
+                )
+                continue
         has_optwarning = False
         for w in ws:
+            # Do not count reaching `maxiter` as an optimization failure/
+            if "ITERATIONS REACHED LIMIT" in str(w.message):
+                continue
```

Comment:
Should we log this on the DEBUG level? It might be good for users to know when their optimization doesn't converge but ends with hitting the iteration limit.